lib/telemetry_influx_db.ex

```diff
@@ -140,13 +149,36 @@ defmodule TelemetryInfluxDB do
     raise(ArgumentError, "protocol has to be :udp or :http")
   end
 
-  defp validate_db!(%{protocol: :udp} = opts), do: opts
-  defp validate_db!(%{protocol: :http, db: _db} = opts), do: opts
+  defp validate_version_params!(%{version: :v2} = opts), do: validate_v2_params!(opts)
+  defp validate_version_params!(%{version: :v1} = opts), do: validate_v1_params!(opts)
 
-  defp validate_db!(_) do
-    raise(ArgumentError, "for http protocol you need to specify :db field")
+  defp validate_version_params!(_opts) do
+    raise(
+      ArgumentError,
+      "version must be :v1 or :v2"
+    )
   end
 
+  defp validate_v2_params!(%{protocol: :udp}) do
+    raise(
+      ArgumentError,
+      "the udp protocol is not currently supported for InfluxDB v2; please use http instead"
+    )
+  end
```

Comment:
Does it make sense to place this clause second? Happy path first, error conditions second in order of specificity?

Reply:
Makes sense to me.